src/main/java/ch/hevs/cloudio/endpoint/CloudioPersistence.java

```diff
@@ -0,0 +1,99 @@
+package ch.hevs.cloudio.endpoint;
+
+import java.util.Set;
+
+public interface CloudioPersistence {
+
+    /**
+     * Open the persistence database
+     */
+    void openDatabase();
+
+    /**
+     * Close the persistence database
+     */
+    void closeDatabase();
+
+    /**
+     * Set a property in the persistence database
+     *
+     * @param key key reference of the property to set
+     * @param value value of the property to set
+     */
+    void setPersistentProperty(String key, Object value);
+
+    /**
+     * Get a property from the persistence database
+     *
+     * @param key key reference of the property to get
+     * @param defaultValue default value to return
+     * @return property from persistence database
+     */
+    Object getPersistentProperty(String key, Object defaultValue);
+
+    /**
+     * Get data of update message from the persistence database
+     *
+     * @param key key reference of the update message to get
+     * @return update message from persistence database
+     */
+    byte[] getPersistentUpdate(String key);
```

Comment:
A persistence layer does not mandatory be based on hash maps, so we should make the interface more general like this:

```java
class CloudioPersistance {
    // ...
    static class Message {
        private long timestamp;
        private String topic;
        private byte[] payload;
    }

    // ...

    void storeMessage(String category, Message message);
    long messageCount(String category);
    Message retrieveNextMessage(String category);
}
```

I did not add getter and setters for Message class for simplicity. The category could represent the different types of messages to store (@update, @log), so we could add stored messages without the need to modify the persistence layer...

Reply:
I'll work on that this week.

Reply:
Did those modifications in 8acb4f6